### ISSUE TITLE
Array size doubles when specifying periodic axes

### ISSUE THREAD

**Issue by mfschubert** [NanoComp/imageruler#24]
There is some unexpected behavior when `periodic_axes` is specified in the `length_violations` function. Specifically, the violations shape may not match the shape of the input array. Example:
```python
print(f"Density shape is {density.shape}")
violations = imageruler.length_violation(density, diameter=9, periodic_axes=None)
print(f"Violations shape with no periodic axes is {violations.shape}")
violations = imageruler.length_violation(density, diameter=9, periodic_axes=(0, 1))
print(f"Violations shape with periodic axes is {violations.shape}")
```
Prints the following:
```
Density shape is (400, 400)
Violations shape with no periodic axes is (400, 400)
Violations shape with periodic axes is (800, 800)
```

**Comment by ianwilliamson** [comment]
Possibly due to the tiling that is applied here: 
https://github.com/NanoComp/imageruler/blob/d77c0afe3db2703dcb741d6b267ceca9678c64cb/imageruler/imageruler.py#L546-L552
but not(?) undone before returning the violations

**Comment by mawc2019** [comment]
This feature is expected and it did not incur an error. To visualize lengthscale violation of a periodic structure, it does no harm to double this structure along the periodic direction(s).

> but not(?) undone before returning the violations

It is not undone.

**Comment by ianwilliamson** [comment]
> This feature is expected and it did not incur an error. To visualize lengthscale violation of a periodic structure, it does no harm to double this structure along the periodic direction(s).

It is not expected because the function's doc string states that the returned quantity "indicates violations in the input image:"

https://github.com/NanoComp/imageruler/blob/d77c0afe3db2703dcb741d6b267ceca9678c64cb/imageruler/imageruler.py#L371

A straightforward interpretation is that the mapping is one-to-one. But, if the output array has a different shape than the input array, it is unclear how a user should interpret the quantities in the output array. No documentation or mapping is provided for how violations may be mapped back to features in the input array.

**Comment by stevengj** [comment]
Yeah, I don't think it makes sense for the function to silently tile the image. 

I agree that tiling the image of a periodic structure can be very helpful for visualization, but the user can do this themselves (or there can be a function provided to do this — it's more flexible to provide additional features by composing multiple functions than by adding lots of options to a single function).

**Comment by stevengj** [comment]
Closed by #30
